core/src/test/java/org/elasticsearch/cluster/metadata/IndexCreationTaskTests.java

```diff
@@ -0,0 +1,450 @@
+    private final AllocationService allocationService = mock(AllocationService.class);
+    private final MetaDataCreateIndexService.IndexValidator validator = mock(MetaDataCreateIndexService.IndexValidator.class);
+    private final ActionListener listener = mock(ActionListener.class);
+    private final ClusterState state = mock(ClusterState.class);
+    private final Settings.Builder clusterStateSettings = Settings.builder();
+    private final MapperService mapper = mock(MapperService.class);
+
+    private final ImmutableOpenMap.Builder<String, IndexTemplateMetaData> tplBuilder = ImmutableOpenMap.builder();
+    private final ImmutableOpenMap.Builder<String, MetaData.Custom> customBuilder = ImmutableOpenMap.builder();
+    private final ImmutableOpenMap.Builder<String, IndexMetaData> idxBuilder = ImmutableOpenMap.builder();
+
+    private final Settings.Builder reqSettings = Settings.builder();
+    private final Set<ClusterBlock> reqBlocks = Sets.newHashSet();
+    private final MetaData.Builder currentStateMetaDataBuilder = MetaData.builder();
+    private final ClusterBlocks currentStateBlocks = mock(ClusterBlocks.class);
+    private final RoutingTable.Builder routingTableBuilder = RoutingTable.builder();
+    private final DocumentMapper docMapper = mock(DocumentMapper.class);
+
+    private ActiveShardCount waitForActiveShardsNum = ActiveShardCount.DEFAULT;
+
+    public void setUp() throws Exception {
+        super.setUp();
+        setupIndicesService();
+        setupClusterState();
+    }
+
+    public void testMatchTemplates() throws Exception {
+        tplBuilder.put("template_1", createTemplateMetadata("template_1", "te*"));
+        tplBuilder.put("template_2", createTemplateMetadata("template_2", "tes*"));
+        tplBuilder.put("template_3", createTemplateMetadata("template_3", "zzz*"));
+
+        final ClusterState result = executeTask();
+
+        assertTrue(result.metaData().index("test").getAliases().containsKey("alias_from_template_1"));
+        assertTrue(result.metaData().index("test").getAliases().containsKey("alias_from_template_2"));
+        assertFalse(result.metaData().index("test").getAliases().containsKey("alias_from_template_3"));
+    }
+
+    public void testApplyDataFromTemplate() throws Exception {
+        addMatchingTemplate(builder -> builder
+                .putAlias(AliasMetaData.builder("alias1"))
+                .putMapping("mapping1", createMapping())
+                .putCustom("custom1", createCustom())
+                .settings(Settings.builder().put("key1", "value1"))
+        );
+
+        final ClusterState result = executeTask();
+
+        assertTrue(result.metaData().index("test").getAliases().containsKey("alias1"));
+        assertTrue(result.metaData().index("test").getCustoms().containsKey("custom1"));
+        assertEquals("value1", result.metaData().index("test").getSettings().get("key1"));
+        assertTrue(getMappingsFromResponse().containsKey("mapping1"));
```

Comment:
Let's replace the `assertTrue` by more effective matchers, and replace the `assertEquals` by `assertThat(..., equalTo(...))`.